commit 82a991a45fb43ea6e03f8f58a197cf3a458b3115
Author: Sven Roederer <devel-sven@geroedel.de>
Date:   Tue Oct 16 21:18:56 2018 +0200

    tools: add zip
    
    it is required by mksercommfw
    
    Signed-of-by

diff --git a/tools/Makefile b/tools/Makefile
index d2b5daf18c..ddc78e49d6 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -31,6 +31,7 @@ tools-$(CONFIG_TARGET_orion_generic) += wrt350nv2-builder upslug2
 tools-$(CONFIG_TARGET_mxs) += elftosb sdimage
 tools-$(CONFIG_TARGET_ar71xx) += lzma-old squashfs
 tools-$(CONFIG_TARGET_ar71xx)$(CONFIG_TARGET_ath79) += squashfs
+tools-$(CONFIG_TARGET_ramips) += zip
 tools-$(CONFIG_USES_MINOR) += kernel2minor
 tools-y += lzma squashfs4
 tools-$(BUILD_B43_TOOLS) += b43-tools
diff --git a/tools/zip/Makefile b/tools/zip/Makefile
new file mode 100644
index 0000000000..a7164aac80
--- /dev/null
+++ b/tools/zip/Makefile
@@ -0,0 +1,36 @@
+# 
+# Copyright (C) 2006-2015 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=zip
+PKG_VERSION:=3.0
+DL_FILE=zip30.tar.gz
+
+PKG_SOURCE:=$(DL_FILE)
+#PKG_SOURCE_URL:=@SF/lzmautils \
+#		http://tukaani.org/xz
+PKG_SOURCE_URL:=@SF/infozip
+PKG_HASH:=f0e8bb1f9b7eb0b01285495a2699df3a4b766784c1765a8f1aeedf63c0806369
+#PKG_BUILD_DIR=zip30
+
+HOST_BUILD_PARALLEL:=0
+HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/zip30
+
+include $(INCLUDE_DIR)/host-build.mk
+
+HOSTCC := $(HOSTCC_NOCACHE)
+HOSTCXX := $(HOSTCXX_NOCACHE)
+
+define Host/Compile
+	+$(MAKE) $(HOST_JOBS) -C $(HOST_BUILD_DIR) -f unix/Makefile generic
+endef
+
+define Host/Install
+	+$(MAKE) $(HOST_JOBS) prefix=$(STAGING_DIR_HOST) -C $(HOST_BUILD_DIR) -f unix/Makefile install
+endef
+
+$(eval $(call HostBuild))
